### PR TITLE
fix: save component version with deploy command

### LINKED ISSUE #2561: Block node add command does not respect release tag set during network deploy
When I execute the following command I spotted some issues.

`solo block node add --deployment "${SOLO_DEPLOYMENT}" --cluster-ref kind-"${SOLO_CLUSTER_NAME}"`

1. Even though I have executed `solo network deploy` in a previous command, using `--release-tag`, the block node sticked on using the default consensus tag for the given solo version.
2. It seems mandatory to explicitly pass `--release-tag` with the consensus node tag you have already deployed
3. The `--release-tag` flag might misleadingly be used for pointing to a block node release tag, instead of the consensus node tag.
4. You can try to deploy a block node even though you might not have performed `solo network deploy` beforehand.

Possible solutions:

1. Rename `--release-tag` to a more elaborate name like `--network-node-release-tag` or adapt block node to respect the network node release tag explicitly set in previous commands during solo deploy. If no `--release-tag` has been set, then it can use the default one for the solo version
2. Forbid block node to be added with a fail fast check that explicitly say network node should have previously been deployed

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

This pull request changes the following:

* Currently, any command would start with `loadAndValidate` step, which would save any chart version into remote config, even before and validation is check again version string, or before the deploy command succeeds.
* Then with the following command, the node start would write default string to remote config again, overwrite the actually version `v0.65.0-rc.3`

```
npm run solo-test -- consensus network deploy --deployment "${SOLO_DEPLOYMENT}" -i node1,node2 --release-tag v0.65.0-rc.3
npm run solo-test -- consensus node setup -i node1,node2 --deployment "${SOLO_DEPLOYMENT}" --release-tag v0.65.0-rc.3
npm run solo-test -- consensus node start -i node1,node2 --deployment "${SOLO_DEPLOYMENT}"
```

* This need to be changed to let individual deploy command to save version info in remote config

### Related Issues

* Closes #2561 
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: src/business/runtime-state/config/remote/remote-config-runtime-state.ts [hunk 1]
```diff
@@ -373,55 +373,60 @@ export class RemoteConfigRuntimeState implements RemoteConfigRuntimeStateApi {
...
-      remoteConfig.versions.explorerChart = new SemVer(argv[flags.explorerVersion.name]);
-    } else if (command === 'explorer' && subcommand === 'node' && action === 'add') {
-      remoteConfig.versions.explorerChart = new SemVer(flags.explorerVersion.definition.defaultValue as string);
-    }
-
-    if (argv[flags.relayReleaseTag.name]) {
-      remoteConfig.versions.jsonRpcRelayChart = new SemVer(argv[flags.relayReleaseTag.name]);
-    } else if (command === 'relay' && subcommand === 'node' && action === 'add') {
-      remoteConfig.versions.jsonRpcRelayChart = new SemVer(flags.relayReleaseTag.definition.defaultValue as string);
+    remoteConfig.versions.chart = argv[flags.soloChartVersion.name]
```
Comment (instamenta) [inline src/business/runtime-state/config/remote/remote-config-runtime-state.ts:376]: I think this will only initialize versions if they are 0.0.0. If intended rename to something like `initializeComponentVersionsInRemoteConfig`

**Review by instamenta — DISMISSED** [review]

File: src/business/runtime-state/config/remote/remote-config-runtime-state.ts (hunk 1 above)
Comment (JeffreyDallas) [reply src/business/runtime-state/config/remote/remote-config-runtime-state.ts:376]: renamed

**Review by instamenta — APPROVED** [review]
